docs/site/pages/plugin-implementation.mdx

```diff
@@ -0,0 +1,94 @@
+---
+title: Plugin Implementation
+---
+
+# Plugin Implementation
+
+The main purpose of a Plugin is to extend or add new functionality by tapping into Player's pipeline of components via hooks. In this section we'll go over the steps to implement a plugin.
+
+We'll use the [stage-revert-data](./plugins/stage-revert-data) plugin as example. After creating our plugin class, we'll use the `apply` method which provides access to the Player instance, which then gives access to the necessary hooks. First we define the variables that will be used in the plugin when accessing the hooks:
```

Comment:
> First we define the variables that will be used in the plugin when accessing the hooks
I would call out why we're doing this since its not something you need to do, but in this case we need to store references in a shared scope.